ps_banner.php

```diff
@@ -58,9 +63,31 @@ public function install()
             $this->registerHook('displayHome') &&
             $this->registerHook('actionObjectLanguageAddAfter') &&
             $this->installFixtures() &&
+            $this->uninstallPrestaShop16Module() &&
             $this->disableDevice(Context::DEVICE_MOBILE));
     }
 
+    /**
+     * Migrate data from 1.6 equivalent module (if applicable), then uninstall
+     */
+    public function uninstallPrestaShop16Module()
+    {
+        if (!Module::isInstalled(self::PS_16_EQUIVALENT_MODULE)) {
+            return true;
+        }
+
+        // Data migration
+        Configuration::updateValue('BANNER_IMG',Configuration::getInt('BLOCKBANNER_IMG'));
+        Configuration::updateValue('BANNER_LINK', Configuration::getInt('BLOCKBANNER_LINK'));
+        Configuration::updateValue('BANNER_DESC', Configuration::getInt('BLOCKBANNER_DESC'));
```

Comment:
This will not work, first of all data from old module is multi-lang, secondly using `getInt` is wrong obviously, I'd go for something like:

`UPDATE `ps_configuration` SET `name` = "BANNER_IMG" WHERE `name` = "BLOCKBANNER_IMG"`

Reply:
@kpodemski It's actually working and as strange as it is, `Configuration::getInt()` is the right method to call here 😅
You can see its definition [here](https://github.com/PrestaShop/PrestaShop/blob/1.7.7.x/classes/Configuration.php#L257-L274).